Model/PricePrecisionConfigTrait.php

```diff
@@ -26,7 +26,11 @@ public function getConfig()
     public function getPricePrecision()
     {
         if ($this->getConfig()->canShowPriceDecimal()) {
-            return $this->getConfig()->getPricePrecision();
+        		//return $context->getLayout();
+        		$Val = $this->getConfig()->getPricePrecision(); 
+        		$Val =  (strstr($_SERVER['PHP_SELF'], 'checkout') > -1 ) ? $this->getConfig()->getPricePrecisionCheckout() :  $Val;
```

Comment:
Don't use global variables.

Reply:
I hope to find the proper variable or object to this.  Meanwhile, I'm looking for the solution, please accept this way. I don't want to steal your project.  I just want to contribute; that's why I don't want nor try to bring users to my fork, but give them as soon as possible a solution ( even if it's not perfect ).  During that time, we'll find the proper way to achieve this code perfectly.